### ISSUE TITLE
FilePicker reports NPE crashes in Google Play Console

### ISSUE THREAD

**Issue by Arcturus5404** [miguelpruivo/flutter_file_picker#397]
**Describe the bug**
Google Play Console reports the following:

```
java.lang.NullPointerException

java.lang.NullPointerException: 
  at com.mr.flutter.plugin.filepicker.FilePickerDelegate$2.run (FilePickerDelegate.java:128)
  at java.lang.Thread.run (Thread.java:919)
```

**Issue details**
1. Platform Android
4. Detailed steps to reproduce:
I currently have no way to reproduce this issue. It works on all emulators I have tested, so I have no idea how serious this problem is.

**Error Log**
None

**Screenshots and/or video**
None

**Flutter Version details**
 flutter --version
Flutter 1.20.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision fba99f6cf9 (3 days ago) • 2020-09-14 15:32:52 -0700
Engine • revision d1bc06f032
Tools • Dart 2.9.2

**Additional context**
Version
+ file_picker 2.0.0

**Comment by miguelpruivo** [comment]
@Arcturus5404 hi! Thank you for reporting. 

This is quite odd actually, it’s throwing a null pointer on an array method in a place where it can’t ever be null (a few lines before it’s instanciated). Could you please post full stack trace?

Thank you!

**Comment by Arcturus5404** [comment]
That's all I have in Google Play Console I'm afraid..

One other interesting detail: I have 15 crash reports
11 on Samsung Galaxy A10e Android 10 (SDK 29)
1 on LGE QStylus Android 8.1 (SDK 27)
3 on Huawei P20 Android 10 (SDK 29)

Not sure of that is of any matter of that its just 1 user trying several times in case of Samsung / Huawei

**Comment by Arcturus5404** [comment]
![image](https://user-images.githubusercontent.com/2432225/93717424-0bbefc80-fb76-11ea-8efb-dd65265354f8.png)

**Comment by Arcturus5404** [comment]
![image](https://user-images.githubusercontent.com/2432225/93717457-50e32e80-fb76-11ea-8883-6a2dab8772c6.png)

**Comment by Arcturus5404** [comment]
I checked your file as well.. can never be null.. is there something weird with my code perhaps?

```
FilePickerResult images =
                          await FilePicker.platform.pickFiles(type: FileType.image, allowMultiple: true);

                      if (images != null && images.paths.isNotEmpty) {
                        List<File> files = images.paths.map((path) => File(path)).toList();
                        setState(() => _stream = chooseAttachments(files));
                      }
```

**Comment by miguelpruivo** [comment]
Does those crashes happen with apps that are using file picker 2.0.0 as well?

**Comment by Arcturus5404** [comment]
Yes! Those are with 2.0.0 file_picker

**Comment by miguelpruivo** [comment]
@Arcturus5404 which version of Flutter are you at?

**Comment by DEMndz5** [comment]
Hello, I've been having these same reports since launch (2 months ago).
I'm now using file_picker 1.13.3, flutter 1.20.4 and the crashes continue.
I don't know if this could be related but I managed to reproduce a crash following these steps:
https://github.com/flutter/flutter/issues/17950
Settings->Developer options->Don't keep activities

![image](https://user-images.githubusercontent.com/62218601/93769477-34380c80-fbf1-11ea-9938-d4d6592d6c66.png)

**Comment by miguelpruivo** [comment]
@DEMndz5 what are the steps to reproduce the issue? After picking?

**Comment by DEMndz5** [comment]
Yes, the app crashes right after picking.

**Comment by miguelpruivo** [comment]
@DEMndz5 and you get a null pointer just like the OP?

**Comment by DEMndz5** [comment]
Yes, this is what I'm getting on the console:
`E/AndroidRuntime( 5737): FATAL EXCEPTION: Thread-6
E/AndroidRuntime( 5737): Process: io.kocho, PID: 5737
E/AndroidRuntime( 5737): java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
E/AndroidRuntime( 5737): 	at com.mr.flutter.plugin.filepicker.FilePickerDelegate$2.run(FilePickerDelegate.java:111)
E/AndroidRuntime( 5737): 	at java.lang.Thread.run(Thread.java:919)`

**Comment by miguelpruivo** [comment]
@DEMndz5 your issue looks different (it might actually be a bug) as it relates to a different line. How are you picking the files?

PS: strangely, it references line 111 which is a blank line on latest version (only 112 has a `equals()` method).

**Comment by DEMndz5** [comment]
I haven't updated to 2.0.0. I can try that later and post the results.
My issue is actually the same as OP but I was trying to reproduce the issue.

**Comment by miguelpruivo** [comment]
@DEMndz5 I suspect you are using 1.13.2.

Could you please try with 2.0.0 (you'll have to refactor a bit of code, but it's worth it) and let me know if it still happens, and if so, which line (perhaps open an issue with that)?

Thank you.

**Comment by miguelpruivo** [comment]
@Arcturus5404 I don't see nothing wrong with your code. Are you able to replicate it with a sample app?

**Comment by Arcturus5404** [comment]
> @Arcturus5404 I don't see nothing wrong with your code. Are you able to replicate it with a sample app?

No I am not I'm afraid.. I am on the following flutter version:

```
== Use Flutter channel stable ==
> flutter --version
Flutter 1.20.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision fba99f6cf9 (3 days ago) • 2020-09-14 15:32:52 -0700
Engine • revision d1bc06f032
Tools • Dart 2.9.2
```

**Comment by Arcturus5404** [comment]
Maybe the issues are related from @DEMndz5, as I build my apks in release / app bundles? Maybe stuff gets confiscated when reports are being sent to the play console?

**Comment by miguelpruivo** [comment]
@Arcturus5404 any news on this?

**Comment by Arcturus5404** [comment]
No nothing.. I think its only happening with a few of my users (3 to be precise).

![image](https://user-images.githubusercontent.com/2432225/94278463-2107a280-ff4b-11ea-876d-d60fe1a84124.png)

I will close the issue, and report back once it becomes a bigger issue!

**Comment by miguelpruivo** [comment]
@Arcturus5404 probably some of those OEM Android ROMS that might be interfering somehow with the correct activity lifecycle. It's not the first time that this happens, specially if those errors happens all on the same devices.

**Comment by Arcturus5404** [comment]
Probably yes! I will keep an eye out on the crash monitor! :) Thanks for your time! Much appreciated

**Comment by miguelpruivo** [comment]
@Arcturus5404 this should be fixed with 2.0.5. It turns out that if some devices destroyed activity while picking files, it could indeed result in a NPE. This was fixed today regarding #402.
